hera_cal/lstbin_simple.py

```diff
@@ -0,0 +1,1740 @@
+
+    logger.info(f"Data Shape: {data.shape}")
+
+    # Now, rephase the data to the lst bin centres.
+    if rephase:
+        logger.info("Rephasing data")
+
+        # lst_mask is a boolean mask that masks out LSTs that are not in any bin
+        # we don't want to spend time rephasing data outside our LST range completely,
+        # so we just mask them out here. Indexing by a boolean mask makes a *copy*
+        # of the data, so we can rephase in-place without worrying about overwriting
+        # the original input data. 
+        data = data[lst_mask]
+        flags = flags[lst_mask]
+        nsamples = nsamples[lst_mask]
+        data_lsts = data_lsts[lst_mask]
+        grid_indices = grid_indices[lst_mask]
+
+        if freq_array is None or antpos is None:
+            raise ValueError("freq_array and antpos is needed for rephase")
+
+        bls = np.array([antpos[k[0]] - antpos[k[1]] for k in antpairs])
+
+        # get appropriate lst_shift for each integration, then rephase
+        lst_shift = lst_bin_centres[grid_indices] - data_lsts
+
+        # this makes a copy of the data in d
+        utils.lst_rephase_vectorized(data, bls, freq_array, lst_shift, lat=lat, inplace=True)
+
+    # In case we don't rephase, the data/flags/nsamples arrays are still the original
+    # input arrays. We don't mask out the data outside the LST range, because we're
+    # just going to omit it from our bins naturally anyway. We also don't care if its
+    # not a copy here, because we're not going to modify it, and this saves memory.
+
+    # We anyway end up with a ~full copy of the data in the output arrays, because
+    # we do a fancy-index of the input arrays to get the relevant data for each bin.
+
+    # TODO: we should think a little more carefully about how we might reduce the 
+    #       number of copies made in this function. When rephasing, we essentially
+    #       get three full copies while inside the function (though one is only local
+    #       in scope, and is therefore removed when the function returns).
+
+    # shortcut -- just return all the data, re-organized.
+    _data, _flags, _nsamples = [], [], []
+    empty_shape = (0, len(antpairs), len(freq_array), npols)
+    for lstbin in range(len(lst_bin_centres)):
+        mask = grid_indices == lstbin
+        if np.any(mask):
+            _data.append(data[mask])
+            _flags.append(flags[mask])
+            _nsamples.append(nsamples[mask])
+        else:
+            _data.append(np.zeros(empty_shape, complex))
+            _flags.append(np.zeros(empty_shape, bool))
+            _nsamples.append(np.zeros(empty_shape, int))
+
+    return lst_bin_centres, _data, _flags, _nsamples
+
+def get_lst_bins(lsts: np.ndarray, edges: np.ndarray) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Get the LST bin indices for a set of LSTs.
+    
+    Parameters
+    ----------
+    lsts
+        The LSTs to bin, in radians.
+    edges
```

Comment:
I can certainly do that, though I disagree with the premise here: they're called `lst_bin_edges` in other functions because otherwise it would not be clear what the edges were referring to. In this function the context makes it clear what edges they are.